lib/settings/settings_notifier.dart

```diff
@@ -0,0 +1,141 @@
+    });
+
+    if (notify) notifyListeners();
+    database!.updateSettings(newSettings: listSettings);
+  }
+
+  ///loads settings from the database. Returns true if managed, else returns false
+  Future<bool> loadSettingsFromDatabase() async {
+    try {
+      List<Setting> databaseSettings = (await database!.getSettings());
+      //get a list of settings keys in database
+      List<String> databaseKeys =
+          List<String>.filled(databaseSettings.length, '');
+      for (int i = 0; i < databaseKeys.length; i++) {
+        databaseKeys[i] = databaseSettings[i].key;
+      }
+
+      for (String key in applicationSettingKeys.keys) {
+        int index = databaseKeys.indexOf(key);
+
+        if (index == -1) {
+          //database is in improper format if does not contain all settings
+          return false;
+        }
+
+        Setting tempSetting = databaseSettings[index];
+        //convert saved string value in database to specified data types
+        switch (applicationSettingKeys[key]) {
+          case 'bool':
+            if (tempSetting.value == 'TRUE') {
+              currentSettings[key] = true;
+            } else if (tempSetting.value == 'FALSE') {
+              currentSettings[key] = 'FALSE';
+            } else {
+              throw Exception(
+                  'Unexpected value for setting $key. Should be of type ${applicationSettingKeys[key]}');
+            }
+            break;
+          case 'bool?':
+            if (tempSetting.value == 'TRUE') {
+              currentSettings[key] = true;
+            } else if (tempSetting.value == 'FALSE') {
+              currentSettings[key] = 'FALSE';
+            } else if (tempSetting.value == 'null') {
+              currentSettings[key] = null;
+            } else {
+              throw Exception(
+                  'Unexpected value for setting $key. Should be of type ${applicationSettingKeys[key]}');
+            }
+            break;
+          case 'String':
+            currentSettings[key] = tempSetting.value;
+            break;
+          case 'int':
+            currentSettings[key] = int.parse(tempSetting.value);
+            break;
+          case 'int?':
+            if (tempSetting.value == '' || tempSetting.value == 'null') {
+              currentSettings[key] = null;
+            } else {
+              currentSettings[key] = int.parse(tempSetting.value);
+            }
+            break;
+          case 'DateTime':
+            currentSettings[key] = DateTime.parse(tempSetting.value);
+            break;
+          case 'DateTime?':
+            if (tempSetting.value == '' || tempSetting.value == 'null') {
+              currentSettings[key] = null;
+            } else {
+              currentSettings[key] = DateTime.parse(tempSetting.value);
+            }
+            break;
+          default:
+            throw Exception(
+                'Unexpected type [${applicationSettingKeys[key]}] for setting [$key]');
```

Comment:
are all of these types necessary? can we not store everything as an integer and just remember to use a converter depending on the use case

Reply:
Want to make sure settings loaded from db aren't corrupted. This way, I don't need checks when using the values.